core/src/test/scala/integration/kafka/api/EndToEndAuthorizationTest.scala

```diff
@@ -170,7 +171,16 @@ trait EndToEndAuthorizationTest extends IntegrationTestHarness with SaslSetup {
     * Tests the ability of producing and consuming with the appropriate ACLs set.
     */
   @Test
-  def testProduceConsume {
+  def testProduceConsumeViaAssign {
+    testProduceConsume(assign = true)
+  }
+
+  @Test
+  def testProduceConsumeViaSubscribe {
+    testProduceConsume(assign = false)
+  }
+
+  private def testProduceConsume(assign : Boolean) {
```

Comment:
@ijuma thanks, weirdness may be subjective considering the method sonsuming the bool is private but no probs changing style :-)